### PR TITLE
Aggregate context (domain service) interaction blogpost

### PR DESCRIPTION
**Pull request by Ocramius**
Fixes https://github.com/Ocramius/ocramius.github.com/issues/103

### CODE REVIEW

**Comment by codeliner** [comment]
well done! I'm currently writing a walk-through tutorial for prooph. I'll definitely link to your blogpost as I've suggested this way of "injecting dependencies" into aggregate methods several times. I think last time I've discussed it with @danizord.

File: source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+</p>
+
+~~~php
+final class ShoppingCart
+{
+    // ... 
+    
+    public function checkOut(CapturedCreditCardCharge $payment) : void
+    {
+        $this->charge = $charge;
```
Comment (akrabat) [inline source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md]: Where did $charge come from?

File: source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
+---
+layout: post
+title: On Aggregates and their interaction with the external world
+category: PHP
+tags: ["php", "ddd", "cqrs", "event sourcing", "aggregate", "aggregate root", "patterns", "clean code"]
+year: 2017
+month: 1
+day: 25
+published: true
+summary: IDDD/DDD Aggregates often have to talk to the external world somehow: a practical approachs
```
Comment (akrabat) [inline source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md]: s/approaches/approach.

File: source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+description: A practical example of how DDD Aggregates can talk to the external world without the need to "know" about its
```
Comment (akrabat) [inline source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md]: Don't end a sentence with a pronoun. 
Or maybe s/its/it.

Comment (Ocramius) [inline source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md]: I think this is my keyboard gone AWOL - it's a bit broken, and sometimes it just hangs. Pushed, thanks :-)

File: source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    // ... 
+    
+    public function __invoke(CheckOutShoppingCart $command) : void
+    {
+        $cartId = $command->shoppingCart();
+        $charge = $command->charge();
+
+        $shoppingCart = $this->carts->get($cartId);
+        
+        // these guards are injected callables:
```
Comment (akrabat) [inline source/_posts/2017-01-25-on-aggregates-and-external-context-interactions.md]: Maybe note that the guards throw an exception on failure?

**Pull request merged**